rclcpp/src/rclcpp/init_options.cpp

```diff
@@ -33,6 +33,7 @@ InitOptions::InitOptions(rcl_allocator_t allocator)
 InitOptions::InitOptions(const rcl_init_options_t & init_options)
 : init_options_(new rcl_init_options_t)
 {
+  std::lock_guard<std::recursive_mutex> init_options_lock(init_options_mutex_);
```

Comment:
taking a mutex in a constructor doesn't make sense

Reply:
right, fix it.